base/essentials.jl

```diff
@@ -95,6 +95,7 @@ tail(x::Tuple) = argtail(x...)
 
 convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::Tuple{Any, Vararg{Any}}) =
     tuple(convert(tuple_type_head(T),x[1]), convert(tuple_type_tail(T), tail(x))...)
+convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::T) = x
```

Comment:
That's semantically the same, but if we actually use `{T<:Tuple}` rather than `{T<:Tuple{Any,Vararg{Any}}}` Julia considers the method on line 96 to be more specific and picks it over the new one:

```julia
julia> Base.convert{T<:Tuple}(::Type{T}, x::T) = x
convert (generic function with 517 methods)

julia> t = (1,2,3,4,5); T = typeof(t);

julia> @which convert(T,t)
convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T<:Tuple{Any,Vararg{Any}}}, x::Tuple{Any,Vararg{Any}}) at essentials.jl:96

julia> Base.convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T}, x::T) = x
convert (generic function with 518 methods)

julia> @which convert(T,t)
convert{T<:Tuple{Any,Vararg{Any}}}(::Type{T<:Tuple{Any,Vararg{Any}}}, x::T<:Tuple{Any,Vararg{Any}}) at none:1
```